src/Symfony/Component/DependencyInjection/Compiler/PriorityTaggedServiceTrait.php

```diff
@@ -116,24 +116,29 @@ class PriorityTaggedServiceUtil
     /**
      * Gets the index defined by the default index method.
      */
-    public static function getDefaultIndex(ContainerBuilder $container, string $serviceId, string $class, string $defaultIndexMethod, string $tagName, string $indexAttribute): ?string
+    public static function getDefaultIndex(ContainerBuilder $container, string $serviceId, string $class, string $defaultIndexMethod, string $tagName, ?string $indexAttribute): ?string
     {
         if (!($r = $container->getReflectionClass($class)) || !$r->hasMethod($defaultIndexMethod)) {
             return null;
         }
 
+        $messageAttributeMissing = '.';
+        if (null !== $indexAttribute) {
+            $messageAttributeMissing = sprintf(' or tag "%s" on service "%s" is missing attribute "%s".', $tagName, $serviceId, $indexAttribute);
+        }
+
         if (!($rm = $r->getMethod($defaultIndexMethod))->isStatic()) {
-            throw new InvalidArgumentException(sprintf('Either method "%s::%s()" should be static or tag "%s" on service "%s" is missing attribute "%s".', $class, $defaultIndexMethod, $tagName, $serviceId, $indexAttribute));
+            throw new InvalidArgumentException(sprintf('Either method "%s::%s()" should be static', $class, $defaultIndexMethod).$messageAttributeMissing);
```

Comment:
Would be great to have a test that check errors messages with and without attributes

Reply:
I added some :)